GetJobProfiles/CypherCommandRecipes/CreateSkillLabelNodes.recipe.json

```diff
@@ -12,9 +12,9 @@
         {
             "name": "CypherCommand",
             "commands": [
-                "match [token:occupationMatch](n:esco__Skill)[token:whereClause]\r\nwith distinct n\r\nunwind n.skos__prefLabel as prefLabels\r\ncreate (l:SkillLabel:Resource {skos__prefLabel: prefLabels, uri: \"<<contentapiprefix>>/SkillLabel/\" + apoc.create.uuid()})\r\ncreate (n)-[:hasPrefLabel]->(l)",
-                "match [token:occupationMatch](n:esco__Skill)[token:whereClause]\r\nwith distinct n\r\nunwind n.skos__altLabel as altLabels\r\ncreate (l:SkillLabel:Resource {skos__prefLabel: altLabels, uri: \"<<contentapiprefix>>/SkillLabel/\" + apoc.create.uuid()})\r\ncreate (n)-[:hasAltLabel]->(l)",
-                "match [token:occupationMatch](n:esco__Skill)[token:whereClause]\r\nwith distinct n\r\nunwind n.skos__hiddenLabel as hiddenLabels\r\ncreate (l:SkillLabel:Resource {skos__prefLabel: hiddenLabels, uri: \"<<contentapiprefix>>/SkillLabel/\" + apoc.create.uuid()})\r\ncreate (n)-[:hasHiddenLabel]->(l)"
+                "with 4 as hexLength\r\nmatch [token:occupationMatch](n:esco__Skill)[token:whereClause]\r\nwith distinct n, hexLength\r\nunwind n.skos__prefLabel as prefLabels\r\nwith n, prefLabels, hexLength, '0x' + right( n.uri, hexLength) as baseHexCode\r\nwith n, prefLabels, hexLength, baseHexCode, toLower(apoc.text.hexValue( apoc.convert.toInteger(baseHexCode))) as replacementChars, left(right( n.uri, 36),36 - hexLength) as trimmedUri, 0 as zero\r\nwith n, prefLabels, hexLength, replacementChars, baseHexCode, trimmedUri + case size(replacementChars) when hexLength then replacementChars when hexLength  / 2  then '0000' + replacementChars else 1/zero end as newUri\r\ncreate (l:SkillLabel:Resource {skos__prefLabel: prefLabels, uri: \"<<contentapiprefix>>/SkillLabel/\" + newUri})\r\ncreate (n)-[:hasPrefLabel]->(l)",
```

Comment:
The case checking for hexLength / 2 looks odd.
Is it to try and handle the replacement chars starting with '0'?
If so, wouldn't it only handle 2 initial 0's, and not say 3 or 1, and shouldn't it be adding 2 0's?

Could it instead pad the start of the value with 0's then take the right of the string, which should also handle overflow?

Should the same logic be used for the occupation label nodes too?